### ISSUE TITLE
Failing to create query in the demo code.

### ISSUE THREAD

**Issue by umar-entertainer** [o19s/elasticsearch-learning-to-rank#243]
I'm using `ElasticSearch 7.2.0` and `Python 3.5.4`.

I was trying to set up the demo to work but I'm halted with this failed to create query bug in the unmodified code.

The code indexes the tmdb data perfectly with a workaround to use 'ltr' instead of '_ltr' since elasticsearch does not support leading underscores anymore. But when it tries to run es.search() it throws error:

> elasticsearch.exceptions.RequestError: TransportError(400, 'search_phase_execution_exception', 'failed to create query: {\n  "bool" : {\n    "filter" : [\n      {\n        "terms" : {\n          "_id" : [\n            "7555",\n            "1370",\n            "1369",\n            "1368",\n            "136278",\n            "102947",\n            "13969",\n            "61645",\n            "14423",\n            "54156"\n          ],\n          "boost" : 1.0\n        }\n      }\n    ],\n    "should" : [\n      {\n        "sltr" : {\n          "featureset" : "movie_features",\n          "params" : {\n            "keywords" : "rambo"\n          },\n          "boost" : 1.0,\n          "_name" : "logged_featureset"\n        }\n      }\n    ],\n    "adjust_pure_negative" : true,\n    "boost" : 1.0\n  }\n}')

The line that's throwing the error is:
`es.search(index=search_index, body=logQuery)`

Server logs of the error:

```
org.elasticsearch.transport.RemoteTransportException: [Umars-MacBook-Pro.local][127.0.0.1:9300][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.index.query.QueryShardException: failed to create query: {
  "bool" : {
    "filter" : [
      {
        "terms" : {
          "_id" : [
            "7555",
            "1370",
            "1369",
            "1368",
            "136278",
            "102947",
            "13969",
            "61645",
            "14423",
            "54156"
          ],
          "boost" : 1.0
        }
      }
    ],
    "should" : [
      {
        "sltr" : {
          "featureset" : "movie_features",
          "params" : {
            "keywords" : "rambo"
          },
          "boost" : 1.0,
          "_name" : "logged_featureset"
        }
      }
    ],
    "adjust_pure_negative" : true,
    "boost" : 1.0
  }
}
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:309) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:292) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.parseSource(SearchService.java:743) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createContext(SearchService.java:591) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:550) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:353) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:340) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:145) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService$2.doRun(SearchService.java:1052) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.2.0.jar:7.2.0]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
    at java.lang.Thread.run(Thread.java:835) [?:?]
Caused by: java.io.IOException: [.ltrstore] IndexNotFoundException[no such index [.ltrstore]]
    at com.o19s.es.ltr.feature.store.index.Caches.cacheLoad(Caches.java:147) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.Caches.loadFeatureSet(Caches.java:129) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.CachedFeatureStore.loadSet(CachedFeatureStore.java:51) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:167) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:49) ~[?:?]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.addBooleanClauses(BoolQueryBuilder.java:394) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.doToQuery(BoolQueryBuilder.java:380) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$1(QueryShardContext.java:293) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:305) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:292) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.parseSource(SearchService.java:743) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createContext(SearchService.java:591) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:550) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:353) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:340) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:145) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService$2.doRun(SearchService.java:1052) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.2.0.jar:7.2.0]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
    at java.lang.Thread.run(Thread.java:835) ~[?:?]
Caused by: org.elasticsearch.index.IndexNotFoundException: no such index [.ltrstore]
    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:194) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:120) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteSingleIndex(IndexNameExpressionResolver.java:282) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.<init>(TransportSingleShardAction.java:164) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.<init>(TransportSingleShardAction.java:141) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:104) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:63) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:145) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.apply(SecurityActionFilter.java:123) ~[?:?]
    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:143) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:121) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:64) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.node.NodeClient.executeLocally(NodeClient.java:83) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.node.NodeClient.doExecute(NodeClient.java:72) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:394) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:383) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:45) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:52) ~[elasticsearch-7.2.0.jar:7.2.0]
    at com.o19s.es.ltr.feature.store.index.IndexFeatureStore.lambda$internalGet$1(IndexFeatureStore.java:192) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.IndexFeatureStore.getAndParse(IndexFeatureStore.java:171) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.IndexFeatureStore.loadSet(IndexFeatureStore.java:118) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.Caches.lambda$cacheLoad$7(Caches.java:140) ~[?:?]
    at org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433) ~[elasticsearch-7.2.0.jar:7.2.0]
    at com.o19s.es.ltr.feature.store.index.Caches.cacheLoad(Caches.java:139) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.Caches.loadFeatureSet(Caches.java:129) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.CachedFeatureStore.loadSet(CachedFeatureStore.java:51) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:167) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:49) ~[?:?]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.addBooleanClauses(BoolQueryBuilder.java:394) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.doToQuery(BoolQueryBuilder.java:380) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$1(QueryShardContext.java:293) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:305) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:292) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.parseSource(SearchService.java:743) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createContext(SearchService.java:591) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:550) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:353) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:340) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:145) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService$2.doRun(SearchService.java:1052) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.2.0.jar:7.2.0]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
    at java.lang.Thread.run(Thread.java:835) ~[?:?]
[2019-09-04T12:56:26,487][DEBUG][o.e.a.s.TransportSearchAction] [Umars-MacBook-Pro.local] All shards failed for phase: [query]
org.elasticsearch.index.query.QueryShardException: failed to create query: {
  "bool" : {
    "filter" : [
      {
        "terms" : {
          "_id" : [
            "7555",
            "1370",
            "1369",
            "1368",
            "136278",
            "102947",
            "13969",
            "61645",
            "14423",
            "54156"
          ],
          "boost" : 1.0
        }
      }
    ],
    "should" : [
      {
        "sltr" : {
          "featureset" : "movie_features",
          "params" : {
            "keywords" : "rambo"
          },
          "boost" : 1.0,
          "_name" : "logged_featureset"
        }
      }
    ],
    "adjust_pure_negative" : true,
    "boost" : 1.0
  }
}
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:309) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:292) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.parseSource(SearchService.java:743) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createContext(SearchService.java:591) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:550) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:353) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:340) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:145) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.search.SearchService$2.doRun(SearchService.java:1052) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758) [elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.2.0.jar:7.2.0]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
    at java.lang.Thread.run(Thread.java:835) [?:?]
Caused by: java.io.IOException: [.ltrstore] IndexNotFoundException[no such index [.ltrstore]]
    at com.o19s.es.ltr.feature.store.index.Caches.cacheLoad(Caches.java:147) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.Caches.loadFeatureSet(Caches.java:129) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.CachedFeatureStore.loadSet(CachedFeatureStore.java:51) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:167) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:49) ~[?:?]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.addBooleanClauses(BoolQueryBuilder.java:394) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.doToQuery(BoolQueryBuilder.java:380) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$1(QueryShardContext.java:293) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:305) ~[elasticsearch-7.2.0.jar:7.2.0]
    ... 16 more
Caused by: org.elasticsearch.index.IndexNotFoundException: no such index [.ltrstore]
    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:194) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:120) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteSingleIndex(IndexNameExpressionResolver.java:282) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.<init>(TransportSingleShardAction.java:164) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.<init>(TransportSingleShardAction.java:141) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:104) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:63) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:145) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.apply(SecurityActionFilter.java:123) ~[?:?]
    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:143) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:121) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:64) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.node.NodeClient.executeLocally(NodeClient.java:83) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.node.NodeClient.doExecute(NodeClient.java:72) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:394) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:383) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:45) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:52) ~[elasticsearch-7.2.0.jar:7.2.0]
    at com.o19s.es.ltr.feature.store.index.IndexFeatureStore.lambda$internalGet$1(IndexFeatureStore.java:192) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.IndexFeatureStore.getAndParse(IndexFeatureStore.java:171) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.IndexFeatureStore.loadSet(IndexFeatureStore.java:118) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.Caches.lambda$cacheLoad$7(Caches.java:140) ~[?:?]
    at org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433) ~[elasticsearch-7.2.0.jar:7.2.0]
    at com.o19s.es.ltr.feature.store.index.Caches.cacheLoad(Caches.java:139) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.Caches.loadFeatureSet(Caches.java:129) ~[?:?]
    at com.o19s.es.ltr.feature.store.index.CachedFeatureStore.loadSet(CachedFeatureStore.java:51) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:167) ~[?:?]
    at com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:49) ~[?:?]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.addBooleanClauses(BoolQueryBuilder.java:394) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.BoolQueryBuilder.doToQuery(BoolQueryBuilder.java:380) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$1(QueryShardContext.java:293) ~[elasticsearch-7.2.0.jar:7.2.0]
    at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:305) ~[elasticsearch-7.2.0.jar:7.2.0]
    ... 16 more

```

**Comment by softwaredoug** [comment]
Hi @umar-entertainer I will take some time to experiment with the demo in the next few days.

**Comment by softwaredoug** [comment]
I'm not currently able to recreate. Can you run this and confirm the plugin installed succesfully?

```
doug@woz $ curl -XGET http://localhost:9200/_ltr
{"stores":{"_default_":{"store":"_default_","index":".ltrstore","version":2,"counts":{"featureset":1,"model":10}}}
```

**Comment by umar-entertainer** [comment]
@softwaredoug Hi, Doug. Thanks for your response. I was able to resolve it. Being a newbie to ES plugins, I was trying to access the `_ltr` plugin through the `ltr` index. Ergo, the failed to create query.

**Comment by anandvijaymonotype** [comment]
@umar-entertainer I also renamed _ltr to ltr but still have the query issue.. Can you please suggest what should i do?

elasticsearch.exceptions.RequestError: RequestError(400, 'parsing_exception', 'no [query] registered for [sltr]')

**Comment by lasagar** [comment]
> @softwaredoug Hi, Doug. Thanks for your response. I was able to resolve it. Being a newbie to ES plugins, I was trying to access the `_ltr` plugin through the `ltr` index. Ergo, the failed to create query.

I am facing the same issue. What did you do to resolve it

**Comment by lasagar** [comment]
> @softwaredoug Hi, Doug. Thanks for your response. I was able to resolve it. Being a newbie to ES plugins, I was trying to access the `_ltr` plugin through the `ltr` index. Ergo, the failed to create query.

I am facing the same issue. What did you do to resolve it

**Comment by umar-entertainer** [comment]
@anandvijaymonotype @lasagar If your ElasticSearch instance is not accepting _ltr, it's basically not detecting the plugin. Please make sure that the correct version is properly installed. And when the ES instance runs, ensure that it has also loaded the plugin.
Secondly, you need to train the data first, before actually querying it. Another issue I was facing is that I was trying to query the plugin without putting an training data in it. Manually create the training data file and then train the ES instance on it. 
Only after that, you will be able to run LTR with ES. Hope this helps.
